src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -113,5 +113,40 @@ public static ReadOnlySpan<byte> AsBytes<T>(this ReadOnlySpan<T> source)
             int newLength = checked((int)((long)source.Length * Unsafe.SizeOf<TFrom>() / Unsafe.SizeOf<TTo>()));
             return new ReadOnlySpan<TTo>(Unsafe.As<Pinnable<TTo>>(source.Pinnable), source.ByteOffset, newLength);
         }
+
+        /// <summary>
+        /// Searches for the specified value and returns the index of its first occurrence. If not found, returns -1. Values are compared using IEquatable&lt;T&gt;.Equals(T). 
+        /// </summary>
+        /// <param name="span">The span to search.</param>
+        /// <param name="value">The value to search for.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int IndexOf<T>(this Span<T> span, T value)
+            where T:struct, IEquatable<T>
+        {
+            return SpanHelpers.IndexOf<T>(ref span.DangerousGetPinnableReference(), value, 0, span.Length);
```

Comment:
The helper will also be used for the ROSpan versions. Also, the helpers use each other (IndexOf of sequence uses IndexOf of value & SequenceEquals).